gammapy/background/tests/test_reflected.py

```diff
@@ -71,19 +71,19 @@ def test_find_reflected_regions(exclusion_mask, on_region):
     regions = fregions.reflected_regions
     assert len(regions) == 16
 
+    # Test with maximum number of regions
+    fregions.max_region_number = 5
+    fregions.run()
+    regions = fregions.reflected_regions
+    assert len(regions) == 5
+
     # Test with too small exclusion
     small_mask = exclusion_mask.cutout(pointing, Angle("0.2 deg"))
     fregions.exclusion_mask = small_mask
     fregions.run()
     regions = fregions.reflected_regions
-    assert len(regions) == 16
-    assert_quantity_allclose(regions[3].center.icrs.ra, Angle("83.674 deg"), rtol=1e-2)
+    assert len(regions) == 0
```

Comment:
This is not the expected behavior.
Here the `get_by_coord()` returns `False` because we are outside the mask. It considers all points outside as excluded.
The boolean type does not allow to properly deal with `True`, `False` and `Invalid`.

Reply:
A possible option is to revert the mask, to put `True` in excluded region instead of `False`. But this reverts the current logic. And then when working with `Map` this might change again.

Reply:
Ah, OK I see the problem. Indeed the invalid value for the bool type does not generalise...I'll think about it again.

Reply:
Thinking about it, I wonder whether this requires a specific `Mask` object or not.
We really need 3 different informations: `is_included`, `is_excluded`, `is_out`.

What to do with the last one depends on the situation:
- Check if reflected region contains an excluded region, or mask some parts of the data for `RingBackground`:  `is_out` must be equivalent to `is_included`.
- Apply a FoV mask, or use a mask to filter an event list: `is_out` has to be equivalent to `is_excluded`.
The behavior is therefore different depending on the expected usage of a mask. One can probably adapt changing from assigning `is_included` to `True` and  `is_excluded` is set to `False` depending on the desired behaviour. This is more or less what `WcsGeom.regions_mask(regions, is_inside)` is doing with the `is_inside` parameter which sets the pixels inside the `regions` to `True`or `False`. 
But this might prove to be a bit complex to understand though and might lead to errors. A minima it should be better documented.

Another option could be to have a `WcsMapMask` object that would have a `.is_excluded(is_out=False)` method with the `is_out` parameter controlling the expected behavior.

Any opinion? @adonath

Reply:
I've thought a bit about this again and my suggestion would be to introduce a `.footprint` property on the `WCSGeom` object, which is a boolean mask defining the validity of the defined projection. Algorithms have to combine the mask data and the footprint specifically for their application.

For this PR I would suggest to just do the type-cast.